lib/ui/compositing/scene_host.cc

```diff
@@ -180,34 +181,44 @@ SceneHost::SceneHost(fml::RefPtr<zircon::dart::Handle> viewHolderToken,
     view_state_changed_callback_.Set(dart_state, viewStateChangedCallback);
   }
 
-  // This callback will be posted as a task  when the |scenic::ViewHolder|
-  // resource is created and given an id by the raster thread.
-  auto bind_callback = [scene_host = this,
-                        isolate_service_id =
-                            isolate_service_id_](scenic::ResourceId id) {
-    const auto key = SceneHostBindingKey(id, isolate_service_id);
-    scene_host_bindings.emplace(std::make_pair(key, scene_host));
+  // This callback will fire when the |scenic::ViewHolder| Resource is created
+  // and given an id by the raster thread.
+  auto on_create_view = [weak = weak_factory_.GetWeakPtr(),
+                         ui_task_runner =
+                             dart_state->GetTaskRunners().GetUITaskRunner(),
+                         isolate_service_id = isolate_service_id_](
+                            scenic::ResourceId resource_id) {
+    ui_task_runner->PostTask([weak = std::move(weak), isolate_service_id,
+                              resource_id]() {
+      if (weak) {
+        FML_DCHECK(weak->view_holder_id_ == 0);
+        weak->view_holder_id_ = resource_id;
+
+        const auto key = SceneHostBindingKey(resource_id, isolate_service_id);
+        scene_host_bindings.emplace(std::make_pair(key, weak.get()));
+      }
```

Comment:
This should prolly not fail silently, consider at least logging an error if not assert

Reply:
if the weak ptr is null

Reply:
consider assert?

Reply:
Done